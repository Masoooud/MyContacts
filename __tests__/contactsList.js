import React from 'react';
import { act, render } from '@testing-library/react-native';
import ContactListScreen from '../src/scenes/contactsList';

describe('testing ContactList Screen', () => {
  test('render testing', () => {
    jest.useFakeTimers();

    const { toJSON } = render(<ContactListScreen />);

    act(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
