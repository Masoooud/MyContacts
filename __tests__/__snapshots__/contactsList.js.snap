// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ContactList Screen render testing 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
    }
  }
  testID="layout"
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={Object {}}
  >
    <View
      accessibilityHint="Header of the page"
      accessibilityLabel="header"
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 32,
        }
      }
      testID="header-container"
    >
      <View />
      <Text
        style={
          Array [
            Object {
              "fontSize": 32,
              "textAlign": "center",
              "width": "34%",
            },
            Object {
              "fontFamily": "System",
              "fontWeight": "700",
            },
          ]
        }
      >
        Contacts
      </Text>
      <View />
    </View>
  </RCTSafeAreaView>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#f1f1f1",
      }
    }
    testID="contacts-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "marginTop": "20%",
            "padding": 32,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
            }
          }
        >
          No contants Available
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
